server/app/services/applicant/question/Scalar.java

```diff
@@ -132,59 +88,60 @@ public static ImmutableSet<Scalar> getScalars(QuestionType questionType)
       throws InvalidQuestionTypeException, UnsupportedQuestionTypeException {
     switch (questionType) {
       case ADDRESS:
-        return ADDRESS_SCALARS;
+        return ImmutableSet.of(
+            STREET,
+            LINE2,
+            CITY,
+            STATE,
+            ZIP,
+            CORRECTED,
+            LATITUDE,
+            LONGITUDE,
+            WELL_KNOWN_ID,
+            SERVICE_AREA);
       case CURRENCY:
-        return CURRENCY_SCALARS;
+        return ImmutableSet.of(CURRENCY_CENTS);
       case DATE:
-        return DATE_SCALARS;
+        return ImmutableSet.of(DATE, YEAR, MONTH, DAY);
       case EMAIL:
-        return EMAIL_SCALARS;
+        return ImmutableSet.of(EMAIL);
       case FILEUPLOAD:
-        return FILE_UPLOAD_SCALARS;
+        return ImmutableSet.of(FILE_KEY, ORIGINAL_FILE_NAME);
       case ID:
-        return ID_SCALARS;
+        return ImmutableSet.of(ID);
       case NAME:
-        return NAME_SCALARS;
+        return ImmutableSet.of(FIRST_NAME, MIDDLE_NAME, LAST_NAME);
       case NUMBER:
-        return NUMBER_SCALARS;
+        return ImmutableSet.of(NUMBER);
       case TEXT:
-        return TEXT_SCALARS;
-
+        return ImmutableSet.of(TEXT);
       case CHECKBOX: // QuestionTypes with multi-selection
-        return MULTI_SELECT_SCALARS;
-
+        return ImmutableSet.of(SELECTIONS);
       case DROPDOWN: // QuestionTypes with single-selection
       case RADIO_BUTTON:
-        return SINGLE_SELECT_SCALARS;
-
+        return ImmutableSet.of(SELECTION);
       case STATIC:
-        return STATIC_SCALARS;
-
+        return ImmutableSet.of();
       case PHONE:
-        return PHONE_SCALARS;
+        return ImmutableSet.of(PHONE_NUMBER, COUNTRY_CODE);
       case ENUMERATOR: // Enumerator Question does not have scalars like the other question types
         // do.
         throw new InvalidQuestionTypeException("Enumeration questions are handled separately.");
-
       case NULL_QUESTION: // Fallthrough intended
       default:
         throw new UnsupportedQuestionTypeException(questionType);
     }
   }
 
   public static ImmutableSet<Scalar> getMetadataScalars() {
-    return METADATA_SCALARS;
+    return ImmutableSet.of(UPDATED_AT, PROGRAM_UPDATED_IN);
   }
 
   /** A set of Scalars as strings that represent keys where metadata is stored. */
   public static ImmutableSet<String> getMetadataScalarKeys() {
-    if (metadataScalarKeys == null) {
-      metadataScalarKeys =
-          METADATA_SCALARS.stream()
-              .map(Scalar::name)
-              .map(String::toLowerCase)
-              .collect(ImmutableSet.toImmutableSet());
-    }
-    return metadataScalarKeys;
+    return getMetadataScalars().stream()
```

Comment:
I understand the reasoning for others, but for metadataScalarKeys, isn't it better to store it at object level instead of calculating it over and over?

Reply:
I think it's an unnecessary optimization because CPU time is ~free and this is not called at a high frequency. But I can fix the original problem by defining the variable statically, so I'll do that instead.